### PR TITLE
General Quality of Life Edits

### PR DESCRIPTION
**Pull request by mayaraman19**
I am deeply sorry for pushing this right before school starts. This PR comprises of a lot of small edits including:

#### Intro
- Made less wordy

#### First, Neural Networks
- Made less wordy (kinda)
- Added links to slides to take up awkward space

#### Shining a Flashlight on Filters
- The flashlight card was compressed from two cards to one
- "If you guessed a koala..." part was made less wordy (mostly) and added koala image on the side

#### Fully-connected Neural Networks
- removed orphan components, made less wordy but now that I'm looking it still seems wordy

#### How do we use filters to analyze images?
- made less wordy

#### Filter picker
- tried to make it fully fit in viewport for as long as possible

#### Filter sizes
- Turned it into a back/next button situation rather than clicking the buttons

#### Intro
- The intro-panel image still needs to be changed?

#### Shining a Flashlight on Filters
- `e.preventDefault()` still isn't working as intended it seems (as in preventing scrolling behavior on touch screens)!

#### How do we use filters to analyze images?
- Looks kinda terrible on wider screens

#### Filter picker
- sizing of filter matrix/lines/squares are kinda awkward

#### More filters
- Need an actual title

#### General
- We are currently considering a style overhaul

### CODE REVIEW

**Comment by mayaraman19** [comment]
i do not know why it includes 10000 commits

**Comment by mayaraman19** [comment]
I'm trying to get the flashlight component to work on mobile, but it still picks up scrolling behavior, making the flashlight really hard to use because the website scrolls at the same time. I added e.preventDefault(), but I'm not sure if I'm using it in the right place. Would appreciate any input <3

**Comment by krashanoff** [comment]
Hey Maya! Thanks for addressing the feedback given. I can't give a cohesive review at the moment, but I *can* provide some answers to your questions:

> i do not know why it includes 10000 commits

Most of these are merges of your remote tracking branch into your local (i.e., updating your local copy of the repository). Don't worry too much about them, they can be compressed down into a single commit with a squash merge.

> it still picks up scrolling behavior, making the flashlight really hard to use because the website scrolls at the same time.

There are three primary events with mobile touches that you would need to account for in the flashlight: `touchstart`, `touchmove`, and `touchend`. Each of these you would need to call `e.preventDefault()` at the start of to ensure that the scrolling does not occur. Alternatively, you could use CSS [as suggested in this thread](https://stackoverflow.com/a/43275544) to prevent the behavior on the component.

**Comment by mayaraman19** [comment]
I am deeply sorry for pushing this right before school starts. This PR comprises of a lot of small edits including:

#### Intro
- Made less wordy

#### First, Neural Networks
- Made less wordy (kinda)
- Added links to slides to take up awkward space

#### Shining a Flashlight on Filters
- The flashlight card was compressed from two cards to one
- "If you guessed a koala..." part was made less wordy (mostly) and added koala image on the side

#### Fully-connected Neural Networks
- removed orphan components, made less wordy but now that I'm looking it still seems wordy

#### How do we use filters to analyze images?
- made less wordy

#### Filter picker
- tried to make it fully fit in viewport for as long as possible

#### Filter sizes
- Turned it into a back/next button situation rather than clicking the buttons

### Issues/Leftovers:
#### Intro
- The intro-panel image still needs to be changed?

#### Shining a Flashlight on Filters
- `e.preventDefault()` still isn't working as intended it seems (as in preventing scrolling behavior on touch screens)!

#### How do we use filters to analyze images?
- Looks kinda terrible on wider screens

#### Filter picker
- sizing of filter matrix/lines/squares are kinda awkward

#### More filters
- Need an actual title

#### General
- We are currently considering a style overhaul

**Review by malsf21 — COMMENTED** [review]
Did a side-by-side comparison and I think in general, this is a solid improvement in quite a few different areas, nothing to complain about! Going to go over the code in-depth tomorrow; thank you for all your work on this so far Maya!

File: src/components/FilterSizes.js [hunk 1]
```diff
@@ -8,58 +8,100 @@ class FilterSizes extends React.Component {
...
+                    style={{top: val[0], left: val[1]}} 
+                    key={index}>{index+1}</div>
         });
+        let showBack = this.state.index > 0 ? "" : " transparent"; //if false, ONLY show next
+        let showNext = this.state.index < 4 ? "" : " transparent"; // if false, ONLY show back
         return (
         <section className = "blue section">
             <div className="card">
-                <h2>click to see what's up</h2>
+                <h2>Yooooo</h2>
```
Comment (malsf21) [inline src/components/FilterSizes.js:88]: Is this supposed to be here 👀

File: src/components/FilterSizes.js [hunk 2]
```diff
@@ -8,58 +8,100 @@ class FilterSizes extends React.Component {
...
-            </p> </>, "78%", "81%"]
+        let params = [ //style.top, style.left
+            ["90%", "19%"],
+            ["54%", "34.75%"],
+            ["47%", "47%"],
+            ["47%", "61.5%"], 
+            ["78%", "81%"]
         ];
         
         const buttons = params.map((val, index) => {
```
Comment (malsf21) [inline src/components/FilterSizes.js:77]: These aren't buttons anymore, right? Do you mind changing the variable name for clarity then?

File: src/components/FilterSizes.js [hunk 3]
```diff
@@ -8,58 +8,100 @@ class FilterSizes extends React.Component {
     constructor(props) {
         super(props);
         this.state = {
+            paragraphs: [ //text, style.top, style.left
```
Comment (malsf21) [inline src/components/FilterSizes.js:11]: Is there a reason that this isn't a constant? Do the paragraphs change?

File: src/components/Filters.js [hunk 4]
```diff
@@ -99,47 +99,48 @@ class Filters extends React.Component {
     render() {
     return (
         <section className="section blue">
-            <div className="filters-holder flexy">
+            <div className="card">
+                <h2>How do filters make features easier to see?</h2>
+                <p>Select the different filters in this example. What patterns do you see in the filter values?</p>
+            </div>
+            <div className="filters-holder">
                 <img src={amsterdamRegular} alt="black and white amsterdam" width="500px"></img>
```
Comment (malsf21) [inline src/components/Filters.js:107]: Ooh, I didn't catch this earlier, but I would avoid hard-coding the `width` as an HTML attribute - it's confusing when combined with CSS classes!

File: src/components/Filters.js [hunk 5]
```diff
@@ -99,47 +99,48 @@ class Filters extends React.Component {
...
+                                <option value="left sobel">Left sobel</option>
+                                <option value="outline">Outline</option>
+                                <option value="right sobel">Right sobel</option>
+                                <option value="sharpen">Sharpen</option>
+                                <option value="top sobel">Top sobel</option>
+                            </select>
+                        </label>
+                    </form>
                 </div>
+                <img id="filter-applied" src={this.state.src} alt="amsterdam with filter applied" width="500px"></img>
```
Comment (malsf21) [inline src/components/Filters.js:143]: Same comment on hard-coded width

File: src/components/Filters.js [hunk 6]
```diff
@@ -99,47 +99,48 @@ class Filters extends React.Component {
...
+                                <option value="blur">Blur</option>
+                                <option value="bottom sobel">Bottom sobel</option>
+                                <option value="emboss">Emboss</option>
+                                <option value="left sobel">Left sobel</option>
+                                <option value="outline">Outline</option>
+                                <option value="right sobel">Right sobel</option>
+                                <option value="sharpen">Sharpen</option>
+                                <option value="top sobel">Top sobel</option>
+                            </select>
+                        </label>
```
Comment (malsf21) [inline src/components/Filters.js:140]: Also disappointed that I didn't catch this myself earlier, but AFAIK the `<label>` and `<select>` are neighbour tags. See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select

File: src/components/Flashlight.js [hunk 7]
```diff
@@ -7,7 +7,8 @@ import './style/Flashlight.css';
 
 class Flashlight extends React.Component {
 
-    showFlashlight = () => {
+    showFlashlight = e => {
+        e.preventDefault();
         const spotlight = document.getElementById("spotlight");
```
Comment (malsf21) [inline src/components/Flashlight.js:12]: Avoid using direct DOM manipulation when possible, though I know the flashlight component is a bit tricky. Not merge-blocking for now, but something we should refactor!

File: src/components/Flashlight.js [hunk 8]
```diff
@@ -7,7 +7,8 @@ import './style/Flashlight.css';
 
 class Flashlight extends React.Component {
 
-    showFlashlight = () => {
+    showFlashlight = e => {
+        e.preventDefault();
```
Comment (malsf21) [inline src/components/Flashlight.js:11]: I can figure this out with you later (the mobile compat), for now let's table it (but maybe add a `// TODO` comment)

File: src/components/Flashlight.js [hunk 9]
```diff
@@ -16,19 +17,38 @@ class Flashlight extends React.Component {
         img.style.filter = "brightness(100%)";
     }
 
-    moveFlashlight = e => {
-        const spotlight = document.getElementById("spotlight");
+    hoverMoveFlashlight = e => {
+        e.preventDefault();
         const container = document.getElementById("container");
```
Comment (malsf21) [inline src/components/Flashlight.js:22]: Same DOM manipulation comment, `// TODO`

File: src/components/Flashlight.js [hunk 10]
```diff
@@ -16,19 +17,38 @@ class Flashlight extends React.Component {
...
         let x = Math.round(e.clientX - rect.x);
         let y = Math.round(e.clientY - rect.y);
+
+        this.moveFlashlight(x, y);
+    }
+
+    touchMoveFlashlight = e => {
+        e.preventDefault();
+        const container = document.getElementById("container");
+        let rect = container.getBoundingClientRect();
```
Comment (malsf21) [inline src/components/Flashlight.js:35]: Same DOM manipulation comment, `// TODO`

File: src/components/Flashlight.js [hunk 11]
```diff
@@ -16,19 +17,38 @@ class Flashlight extends React.Component {
...
+
+        //get coordinates of mouse
+        let x = Math.round(e.targetTouches[0].clientX - rect.x - 20);
+        let y = Math.round(e.targetTouches[0].clientY - rect.y - 20);
```
Comment (malsf21) [inline src/components/Flashlight.js:39]: Do you have to round it here? IRRC floats are supported with pixel values

File: src/components/Flashlight.js [hunk 12]
```diff
@@ -16,19 +17,38 @@ class Flashlight extends React.Component {
...
+
+        this.moveFlashlight(x, y);
+    }
+
+    moveFlashlight = (x, y) => {
+        const spotlight = document.getElementById("spotlight");
```
Comment (malsf21) [inline src/components/Flashlight.js:45]: Same DOM manipulation comment, `// TODO`

File: src/components/Flashlight.js [hunk 13]
```diff
@@ -16,19 +17,38 @@ class Flashlight extends React.Component {
...
         spotlight.style.left = (x) + 'px';
         spotlight.style.top = (y) + 'px';
     }
 
-    hideFlashlight = () => {
+    hideFlashlight = e => {
+        e.preventDefault();
         const spotlight = document.getElementById("spotlight");
```
Comment (malsf21) [inline src/components/Flashlight.js:52]: Same DOM manipulation comment, `// TODO`

**Review by malsf21 — APPROVED** [review]
Left some minor feedback, but I don't think anything's merge-blocking per se - knock out however much you feel like doing now, but you're always free to punt things to later. If so, just do me a favour and add a `// TODO` comment so we don't forget!

Another note: there is quite a few classes that are heavily duplicated (e.g. `color` or `background-color` for the AI blue, some flex utilities). Not merge-blocking either, but when you do end up doing the redesign - it may be helpful to consolidate some of those classes into reusable components, which should be easier to maintain.

Still, major respect for hitting on so much feedback @mayaraman19! Really appreciate it :) 

(oh, and please squash merge - there's quite a few commits!)

File: src/components/FilterSizes.js (hunk 3 above)
Comment (mayaraman19) [reply src/components/FilterSizes.js:11]: they don't change; i've moved it out of state

**Pull request merged**